app/src/main/java/protect/card_locker/importexport/StocardImporter.java

```diff
@@ -144,18 +142,20 @@ public void importData(Context context, SQLiteDatabase database, File inputFile,
         saveAndDeduplicate(context, database, importedData);
     }
 
-    public ZIPData importZIP(ZipInputStream zipInputStream, final ZIPData zipData) throws IOException, FormatException, JSONException {
+    public ZIPData importZIP(ZipFile zipFile, final ZIPData zipData) throws IOException, FormatException, JSONException {
         Map<String, StocardRecord> cards = zipData.cards;
         Map<String, StocardProvider> providers = zipData.providers;
 
         String[] customProvidersBaseName = null;
         String[] cardBaseName = null;
         String customProviderId = "";
         String cardName = "";
-        LocalFileHeader localFileHeader;
-        while ((localFileHeader = zipInputStream.getNextEntry()) != null) {
-            String fileName = localFileHeader.getFileName();
+        List<FileHeader> fileHeaders = zipFile.getFileHeaders();
+        fileHeaders.sort(Comparator.comparingLong(FileHeader::getOffsetLocalHeader));
```

Comment:
I could also remove this. It should not be needed and deviates from how most programs would read the entries (but only if the order actually differs, which is rare). But since the importer may depend on the current iteration order of the entries I figured it best not to change it unless you want me to.